helmclient.go

```diff
@@ -271,7 +271,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 		}
 	}
 
-	if pod != nil {
+	if !installTiller && pod != nil {
```

Comment:
This fixes a bug I introduced with the tiller upgrade logic. If we are installing tiller there is no point doing the version check.

This was causing this error to be returned in some situations.

`return microerror.Maskf(executionFailedError, "invalid state cannot both install and upgrade tiller")`